### PR TITLE
Make NtfsFileAttributes a nullable flags enum

### LINKED ISSUE #8082: [P1] Storage: Change NtfsFileAttributes to a nullable flags enum
https://apiview.azurewebsites.net/Assemblies/Review/47c9ebbff8c0458fa2e7904eae34656f#Azure.Storage.Files.Models.NtfsFileAttributes

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes #8082 
- Replaced Normal with None as Normal attribute is not mentioned in the docs anywhere.

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
Have you run the related unit tests in live mode?

File: sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs [hunk 1]
```diff
@@ -60,202 +52,151 @@ public class NtfsFileAttributes
...
         /// <summary>
         /// ToString
         /// </summary>
         /// <returns>string</returns>
-        public override string ToString()
+        public static string ToAttributesString(this NtfsFileAttributes attributes)
         {
             var stringBuilder = new StringBuilder();
 
-            if (None)
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs]: You need to keep None here.  null means unspecified, None means the attributes are empty.

File: sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs [hunk 2]
```diff
@@ -60,202 +52,151 @@ public class NtfsFileAttributes
...
+            if ((attributes & NtfsFileAttributes.ReadOnly) == NtfsFileAttributes.ReadOnly)
             {
-                stringBuilder.Append(nameof(None) + "|");
+                stringBuilder.Append(nameof(NtfsFileAttributes.ReadOnly) + "|");
```
Comment (tg-msft) [inline sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs]: Nit - do these as two separate Appends so you're allocating fewer strings.

**Review by tg-msft — APPROVED** [review]

File: sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs]: But none is the default value we pass to the service if the enum is null. Why would someone need to specify None explicitly?

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs]: Is that what we do?  The way I understood it, when it's null we shouldn't pass anything (which leaves the value alone) and when it's `None` we should pass that (which will clear any value that's there).

Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs]: For the SetHttpHeaders it defaults to Preserve, for Create it defaults to None.
I think the Normal enum value is essentially the same as the None value. Should we delete Normal and update description of None? I don't see where Normal is actually referenced in the docs - https://docs.microsoft.com/en-us/rest/api/storageservices/create-directory

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs]: @seanmcc-msft - can you weigh in here?  I remember we discussed [this awhile back](https://github.com/Azure/azure-sdk-for-net/pull/7361#discussion_r316425454) and maybe I'm getting it wrong.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs]: I agree with Josh, null is supposed to mean leave the current NTFS attributes.  We don't need None in the enum.

Comment (JoshLove-msft) [reply sdk/storage/Azure.Storage.Files/src/Models/NtfsFileAttributes.cs]: I replaced Normal with None.

**Comment by JoshLove-msft** [comment]
> Have you run the related unit tests in live mode?

Yes, I ran in live mode.

**Pull request merged**